unit_tests/src/test_atomic_apply_acl_operations.cpp

```diff
@@ -432,6 +433,210 @@ TEST_CASE("#7338")
     CHECK(json_error_string == nullptr);
 }
 
+TEST_CASE("Atomic acls respect entity zone")
+{
+    using namespace std::string_literals;
+
+    rodsEnv env;
+    _getRodsEnv(env);
+    irods::experimental::client_connection conn;
+    const auto user_home = fs::path{env.rodsHome}.string();
+    const auto* const test_zone_name = "atomic_acl_test_zone";
+    const auto* const acl_user_name = "atomic_acl_user";
+    const auto* const remote_only_user_name = "atomic_acl_remote_zone_user";
+
+    const auto* const local_zone_name = env.rodsZone;
+
+    // add zone for testing
+    REQUIRE_NOTHROW(ua::client::add_zone(conn, test_zone_name));
+
+    // clean up zone after end
+    irods::at_scope_exit remove_test_zone{[&conn, &test_zone_name] {
+        if (ua::client::zone_exists(conn, test_zone_name)) {
+            ua::client::remove_zone(conn, test_zone_name);
+        }
+    }};
+
+    // create 3 users:
+    // two users with identical names but differing zones
+    ua::user remote_zone_user{acl_user_name, test_zone_name};
+    ua::user local_zone_user{acl_user_name};
+    // one user with unique name in remote zone only
+    ua::user remote_only_user{remote_only_user_name, test_zone_name};
+
+    // clean up users after end
+    irods::at_scope_exit remove_users{[&] {
+        for (auto&& u : {remote_zone_user, local_zone_user, remote_only_user}) {
+            ua::client::remove_user(conn, u);
+        }
+    }};
+
+    if (!ua::client::exists(conn, remote_zone_user)) {
```

Comment:
Let's remove these checks.

If a requirement of the test isn't satisfied, we need the test to report that. The check for the existence of the user can lead to a prior test getting by without cleaning up.